enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4919,6 +4929,416 @@ class AdjointGenerator
         }
       }
 
+      if (Mode == DerivativeMode::ReverseModeGradient) {
+        eraseIfUnused(call, /*erase*/ true, /*check*/ false);
+      } else {
+        eraseIfUnused(call);
+      }
+    }
+
+    if ((funcName == "cblas_sswap" || funcName == "cblas_dswap") &&
+        called->isDeclaration()) {
+      // sswap(int n, float *x, int incx, float *y, int incy)
+      handled = true;
+      auto arg_n = call.getArgOperand(0), arg_x = call.getArgOperand(1),
+           arg_incx = call.getArgOperand(2), arg_y = call.getArgOperand(3),
+           arg_incy = call.getArgOperand(4);
+      auto has_diff_x = !gutils->isConstantValue(arg_x),
+           has_diff_y = !gutils->isConstantValue(arg_y);
+      if (Mode == DerivativeMode::ReverseModeCombined ||
+          Mode == DerivativeMode::ReverseModeGradient) {
+        IRBuilder<> Builder2(call.getParent());
+        getReverseBuilder(Builder2);
+        if (has_diff_x) {
```

Comment:
Do you not want to check both has_diff_x and has_diff_y?

Reply:
It is checked by the assert statement below